src/test/fuzz/banman.cpp

```diff
@@ -38,51 +38,43 @@ FUZZ_TARGET_INIT(banman, initialize_banman)
         while (fuzzed_data_provider.ConsumeBool()) {
-            switch (fuzzed_data_provider.ConsumeIntegralInRange<int>(0, 11)) {
-            case 0: {
-                ban_man.Ban(ConsumeNetAddr(fuzzed_data_provider),
-                    ConsumeBanTimeOffset(fuzzed_data_provider), fuzzed_data_provider.ConsumeBool());
-                break;
-            }
-            case 1: {
-                ban_man.Ban(ConsumeSubNet(fuzzed_data_provider),
-                    ConsumeBanTimeOffset(fuzzed_data_provider), fuzzed_data_provider.ConsumeBool());
-                break;
-            }
-            case 2: {
-                ban_man.ClearBanned();
-                break;
-            }
-            case 4: {
-                ban_man.IsBanned(ConsumeNetAddr(fuzzed_data_provider));
-                break;
-            }
-            case 5: {
-                ban_man.IsBanned(ConsumeSubNet(fuzzed_data_provider));
-                break;
-            }
-            case 6: {
-                ban_man.Unban(ConsumeNetAddr(fuzzed_data_provider));
-                break;
-            }
-            case 7: {
-                ban_man.Unban(ConsumeSubNet(fuzzed_data_provider));
-                break;
-            }
-            case 8: {
-                banmap_t banmap;
-                ban_man.GetBanned(banmap);
-                break;
-            }
-            case 9: {
-                ban_man.DumpBanlist();
-                break;
-            }
-            case 11: {
-                ban_man.Discourage(ConsumeNetAddr(fuzzed_data_provider));
-                break;
-            }
-            }
+            CallOneOf(
+                fuzzed_data_provider,
+                [&] {
+                    ban_man.Ban(ConsumeNetAddr(fuzzed_data_provider),
+                                ConsumeBanTimeOffset(fuzzed_data_provider), fuzzed_data_provider.ConsumeBool());
+                },
+                [&] {
+                    ban_man.Ban(ConsumeSubNet(fuzzed_data_provider),
+                                ConsumeBanTimeOffset(fuzzed_data_provider), fuzzed_data_provider.ConsumeBool());
+                },
+                [&] {
+                    ban_man.ClearBanned();
+                },
+                [] {},
+                [&] {
+                    ban_man.IsBanned(ConsumeNetAddr(fuzzed_data_provider));
+                },
+                [&] {
+                    ban_man.IsBanned(ConsumeSubNet(fuzzed_data_provider));
+                },
+                [&] {
+                    ban_man.Unban(ConsumeNetAddr(fuzzed_data_provider));
+                },
+                [&] {
+                    ban_man.Unban(ConsumeSubNet(fuzzed_data_provider));
+                },
+                [&] {
+                    banmap_t banmap;
+                    ban_man.GetBanned(banmap);
+                },
+                [&] {
+                    ban_man.DumpBanlist();
+                },
+                [] {},
```

Comment:
?

Reply:
This is needed to preserve the "refactoring" nature of this pull request. Whether to remove or keep the noop should be discussed in #20837 .

Reply:
(Note that case 10 does not exists, i.e. is a noop)